apps/web/app.vue

```diff
@@ -56,4 +56,16 @@ watch(
     await getCategoryTree();
   },
 );
+
+watch(
```

Comment:
We currently write all our watchers on the app.vue level which works fine but it will get crowded in the future. We should aim to move the logic into middlewares and plugins. 

Lets move the code into a middleware and register that middleware in the app.vue or maybe there is a way to globally register that middleware. 

[example middleware](https://github.com/plentymarkets/plentyshop-pwa/blob/5c697c30db0ebfb5763cc326e19b71eaca2bd6cb/apps/web/middleware/reject-empty-checkout.ts)
[how to use the middleware](https://github.com/plentymarkets/plentyshop-pwa/blob/5c697c30db0ebfb5763cc326e19b71eaca2bd6cb/apps/web/pages/checkout.vue)

https://nuxt.com/docs/guide/directory-structure/middleware